### PR TITLE
Updated UI for Edit Visible Assets Modal

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Updated the UI for Edit Visible Assets Modal.
 
1) Moved the `Add custom asset` button to the top left above the list and made it a secondary button.
2) Added primary `Done` button to the bottom of the modal.
3) Fixed an async `error` that was occurring when removing an asset from the list.
4) Added error handling for when adding a new asset that the `pricingAPI` doesn't recognize, will now return a zero balance for that asset instead of failing the entire portfolio price information.
5) Added a timeout method to the `EditVisibleAssets` modal to throttle the `fullAssetList` to allow the modal to display instantly and then load asset data.
6) Added checks in places that were calling the `getAllTokensList` method, to only call if the `fullAssetList` length is `0`

Resolves <https://github.com/brave/brave-browser/issues/18474>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/135563649-d508234e-088a-40b8-b145-ecedf9de6f4c.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx [hunk 1]
```diff
@@ -94,7 +95,11 @@ const EditVisibleAssetsModal = (props: Props) => {
   }, [fullAssetList, userVisibleTokensInfo])
 
   React.useMemo(() => {
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:97]: Is `React.useEffect` more appropriate here?

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx [hunk 2]
```diff
@@ -293,8 +307,8 @@ const EditVisibleAssetsModal = (props: Props) => {
                   </>
                 </WatchlistScrollContainer>
                 <NavButton
-                  onSubmit={toggleShowAddCustomToken}
-                  text={getLocale('braveWalletWatchlistAddCustomAsset')}
+                  onSubmit={onClose}
```
Comment (onyb) [inline components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:310]: Is the Done button doing nothing? I think the expectation from such UX is that if you don't click on Done (and just close the modal), then no changes made will be applied, which doesn't seem to be the case.

Is it possible to make Done behave in a way that applies all the changes in one go? If it's too much work, we can keep it as-is, but in that case we might as well not display the Done button at all.

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx]: Updated!

File: components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:310]: Per slack discussion, currently the api is not set up to handle multiple adding/removing of tokens at once, it has to be handled 1 at a time.

Comment (bbondy) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:310]: Pls post a follow up for the behavior that @onyb suggested.  It's not MVP blocking but what he's suggesting is best.

Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx:310]: Posted an issue here: https://github.com/brave/brave-browser/issues/18532

**Review by onyb — APPROVED** [review]

**Review by bbondy — APPROVED** [review]

**Pull request merged**